apps/settings/src/components/PersonalInfo/EmailSection/EmailSection.vue

```diff
@@ -54,14 +58,14 @@
 <script>
 import { loadState } from '@nextcloud/initial-state'
 import { showError } from '@nextcloud/dialogs'
-import '@nextcloud/dialogs/styles/toast.scss'
 
-import HeaderBar from './HeaderBar'
 import Email from './Email'
-import { savePrimaryEmail, removeAdditionalEmail } from '../../../service/PersonalInfoService'
-import { DEFAULT_ADDITIONAL_EMAIL_SCOPE } from '../../../constants/AccountPropertyConstants'
+import HeaderBar from '../shared/HeaderBar'
 
-const { additionalEmails, primaryEmail } = loadState('settings', 'emails', {})
+import { ACCOUNT_PROPERTY_READABLE_ENUM, DEFAULT_ADDITIONAL_EMAIL_SCOPE } from '../../../constants/AccountPropertyConstants'
+import { savePrimaryEmail, savePrimaryEmailScope, removeAdditionalEmail } from '../../../service/PersonalInfo/EmailService'
+
+const { emails: { additionalEmails, primaryEmail } } = loadState('settings', 'personalInfoParameters', {})
```

Comment:
That could prevent an exception

```suggestion
const { emails: { additionalEmails, primaryEmail } } = loadState('settings', 'personalInfoParameters', { emails: {}})
```

Reply:
Wrote this similarly to other `loadState` calls in the codebase which don't handle errors — with your suggestion other errors will throw further down the chain. I've written up a potential solution which catches `loadState` errors, but would need more input from others to decide whether it's a good solution
```js
// main-personal-info.js
const EmailView = Vue.extend(EmailSection)

try {
	const { emails: { additionalEmails, primaryEmail } } = loadState('settings', 'personalInfoParameters', {})

	new EmailView({
		propsData {
			additionalEmails,
			primaryEmail,
		},
	}).$mount('#vue-emailsection')
} catch (e) {
	logger.error('Could not load personalInfoParameters state', e)
}
```

Reply:
Yes, you are right, it makes sense to let `loadState` throw an error instead of creating more obscure errors down the chain. I don't even think that the extra `try/catch` is necessary.

Reply:
Fail fast 😂, so is it fine with you if I keep it the same and let `loadState` throw if there are errors?

Reply:
Yes :smiley: